libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java

```diff
@@ -0,0 +1,858 @@
+  private static final long GUIDANCE_VIEW_HIDE_TRANSITION_SPEED = 900L;
+
+  private ManeuverView upcomingManeuverView;
+  private TextView upcomingDistanceText;
+  private TextView upcomingPrimaryText;
+  private TextView upcomingSecondaryText;
+  private ManeuverView subManeuverView;
+  private TextView subStepText;
+  private NavigationAlertView alertView;
+  private View rerouteLayout;
+  private View turnLaneLayout;
+  private View subStepLayout;
+  private ImageView guidanceViewImage;
+  private RecyclerView rvTurnLanes;
+  private RecyclerView rvInstructions;
+  private TurnLaneAdapter turnLaneAdapter;
+  private ConstraintLayout instructionLayout;
+  private LinearLayout instructionLayoutText;
+  private View instructionListLayout;
+  private InstructionListAdapter instructionListAdapter;
+  private Animation rerouteSlideUpTop;
+  private Animation rerouteSlideDownTop;
+  private LegStep currentStep;
+  private NavigationViewModel navigationViewModel;
+  private InstructionListListener instructionListListener;
+
+  private DistanceFormatter distanceFormatter;
+  private boolean isRerouting;
+  private SoundButton soundButton;
+  private FeedbackButton feedbackButton;
+  private LifecycleOwner lifecycleOwner;
+
+  public InstructionView(Context context) {
+    this(context, null);
+  }
+
+  public InstructionView(Context context, @Nullable AttributeSet attrs) {
+    this(context, attrs, -1);
+  }
+
+  public InstructionView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
+    super(context, attrs, defStyleAttr);
+    initialize();
+  }
+
+  /**
+   * Adds a listener that is triggered when the instruction list in InstructionView is shown or hidden.
+   *
+   * @param instructionListListener to be set
+   */
+  public void setInstructionListListener(InstructionListListener instructionListListener) {
+    this.instructionListListener = instructionListListener;
+  }
+
+  /**
+   * Once this view has finished inflating, it will bind the views.
+   * <p>
+   * It will also initialize the {@link RecyclerView} used to display the turn lanes
+   * and animations used to show / hide views.
+   */
+  @Override
+  protected void onFinishInflate() {
+    super.onFinishInflate();
+    bind();
+    initializeBackground();
+    initializeTurnLaneRecyclerView();
+    initializeInstructionListRecyclerView();
+    initializeAnimations();
+    initializeStepListClickListener();
+    initializeButtons();
+    ImageCreator.getInstance().initialize(getContext());
+  }
+
+  @Override
+  protected void onAttachedToWindow() {
+    super.onAttachedToWindow();
+    addBottomSheetListener();
+  }
+
+  @Override
+  protected void onDetachedFromWindow() {
+    super.onDetachedFromWindow();
+    cancelDelayedTransition();
+  }
+
+  @Override
+  public void onFeedbackSelected(FeedbackItem feedbackItem) {
+//    navigationViewModel.updateFeedback(feedbackItem); // TODO Telemetry impl
```

Comment:
Let's cut tickets with this `TODO` and the rest 👇 so we don't forget 👍

Reply:
https://github.com/mapbox/mapbox-navigation-android/issues/2507